api/integrations/github/client.py

```diff
@@ -66,9 +68,34 @@ def generate_jwt_token(app_id: int) -> str:  # pragma: no cover
     return token
 
 
+def build_paginated_response(
+    results: list[dict[str, Any]],
+    response: requests.Response,
+    total_count: int | None = None,
+    incomplete_results: bool | None = None,
+) -> dict[str, Any]:
```

Comment:
nit: It's not critical, but, since we fully control the creation of this dict, we could add a `TypedDict` return type to this function that would help introspection.

Reply:
Maybe Dataclass or would it be too much? Let's please discuss this. Merging as is FTM.